pyvista/utilities/geometric_objects.py

```diff
@@ -50,11 +50,12 @@ def translate(surf, center=[0., 0., 0.], direction=[1., 0., 0.]):
 
     surf.transform(trans)
     if not np.allclose(center, [0., 0., 0.]):
-        surf.points += np.array(center)
+        surf.points[:] += np.array(center)
```

Comment:
This change seems superfluous to me, at least I suspect `surf.points` generates a new array each time. What's the reason for it?

Reply:
Good catch.  I've been getting collection issues locally, and `[:]` doesn't cause the same issue.  I have no idea what's going on behind scenes and why a reference is being kept.

Reply:
I can repro the error locally. Unresolving this for now because I want to understand what's going on here.

Reply:
OK, I'm not there yet but I understand half the situation.

I believe the original
```py
        surf.points += np.array(center)
```
is equivalent to
```py
        surf.points = surf.points.__iadd__(np.array(center))
```

whereas the new
```py
        surf.points[:] += np.array(center)
```
is equivalent to
```py
        arr = surf.points[:]
        arr = arr.__iadd__(np.array(center))
        del arr
```

The new version works regardless, because the `__iadd__` call mutates the array, which is a view of the original returned from the property, which in turn directly reference (and thus mutates) the VTK array thanks to `vtk.vtk_to_numpy`.

And the key difference between the two is that the old version calls the setter of `DataSet.points`, whereas the new version does not. So I only have to figure out what reference sticks around when calling that setter from `tests/plotting/test_plotting.py`, and why it doesn't break any  other tests where the setter gets called. I suspect it's the combination of setter and `+=`, but I'm not sure.

Reply:
OK, for the time being I don't know what's causing this, but we should consider spelling this out as
```py
        points = surf.points
        points += np.array(center)
```
just to make it more obvious that it's on purpose. But I'll try to look into this still and we can change this too in a possible future PR...

Reply:
On `master` when running `pytest -vx tests/plotting`, the first run executes fine, but with the second run I get:
```
tests/plotting/test_plotting.py::test_plot_pyvista_ndarray PASSED       [  0%]
tests/plotting/test_plotting.py::test_plot_pyvista_ndarray ERROR        [  0%]

=================================== ERRORS ====================================
_______________ ERROR at teardown of test_plot_pyvista_ndarray ________________

    @pytest.fixture(autouse=True)
    def check_gc():
        """Ensure that all VTK objects are garbage-collected by Python."""
        before = set(id(o) for o in gc.get_objects() if _is_vtk(o))
        yield
        pyvista.close_all()
    
        gc.collect()
        after = [o for o in gc.get_objects() if _is_vtk(o) and id(o) not in before]
>       assert len(after) == 0, \
            'Not all objects GCed:\n' + \
            '\n'.join(sorted(o.__class__.__name__ for o in after))
E       AssertionError: Not all objects GCed:
E         vtkFloatArray
E       assert 1 == 0
E         +1
E         -0

```

I can't fathom how there's a side effect between multiple runs of `pytest`.  Better, when I switch back to this branch, the same thing happens; first test runs fine and second test fails.

Moreover, I can't find a commit in the past 14 months where the following standalone unit test passes:


```py
import pyvista
import pytest
import gc


def _is_vtk(obj):
    try:
        return obj.__class__.__name__.startswith('vtk')
    except Exception:  # old Python sometimes no __class__.__name__
        return False


@pytest.fixture(autouse=True)
def check_gc():
    """Ensure that all VTK objects are garbage-collected by Python."""
    before = set(id(o) for o in gc.get_objects() if _is_vtk(o))
    yield
    pyvista.close_all()
    gc.collect()
    after = [o for o in gc.get_objects() if _is_vtk(o) and id(o) not in before]
    assert len(after) == 0, \
        'Not all objects GCed:\n' + \
        '\n'.join(sorted(o.__class__.__name__ for o in after))


def test_translate_plot():
    plotter = pyvista.Plotter(off_screen=True)

    plane_y = 2  # must be non-zero
    screen = pyvista.Plane(center=(0, plane_y, 0),
                           direction=(0, 1, 0),
                           i_size=5, j_size=5)
    plotter.add_mesh(screen)
    plotter.show()
```

This is a task for another PR.  For now, let's leave in the "hotfix".

https://github.com/pyvista/pyvista/issues/1248